resources/lib/lcdproc.py

```diff
@@ -496,11 +499,11 @@ def SetBigDigits(self, strTimeString, bForceUpdate):
       if self.m_strDigits[iDigitCount] != "" or bForceUpdate:
         self.m_strDigits[iDigitCount] = ""
         self.m_strSetLineCmds += "widget_set xbmc lineBigDigit" + str(iDigitCount) + " 0 0\n"
-      
+
       iDigitCount += 1
 
   def SetProgressBar(self, percent, pxWidth):
-    self.m_iProgressBarWidth = int(float(percent) * pxWidth)
+    self.m_iProgressBarWidth = int((float(percent)/100) * pxWidth)
```

Comment:
? Whats wrong with the progress bar percentage calculation?

Reply:
I think it is not consistent, I'm not sure how should it be though:

From `lcdbase.py`:

``` python
      if self.m_extraBars[i] == "progress":
        if isplaying:
          self.m_cExtraIcons.SetBar(i, (self.m_InfoLabels.GetProgressPercent() * 100))
        else:
          self.m_cExtraIcons.SetBar(i, 0)
      elif self.m_extraBars[i] == "volume":
        self.m_cExtraIcons.SetBar(i, self.m_InfoLabels.GetVolumePercent())
      elif self.m_extraBars[i] == "volumehidden":
        if self.m_bVolumeChangeActive:
          self.m_cExtraIcons.SetBar(i, self.m_InfoLabels.GetVolumePercent())
```

`GetProgressPercent()` and `GetVolumePercent()` return different range for percentage (0-1 range vs. 0-100 range).

Reply:
True about the different ranges/scales. Though, for progress you get 0-1, so one can simply multiply that float value with the number of horizontal pixels (lcdproc knows about that and reports this accordingly), and for 75% progress with a width of 40 pixels (40 * 0.75), you get the correct value of 30 pixels. No need to do further math on this.